### PR TITLE
chore: do not list all snapshots in memory when asked for them

### LINKED ISSUE #7198: Limit RAM usage when Listin snapshots
DO THIS.

### PR DESCRIPTION
**Pull request by jeremyletang**
close #7198

### CODE REVIEW

**Review by EVODelavega — APPROVED** [review]

File: core/snapshot/engine.go [hunk 1]
```diff
@@ -218,6 +219,40 @@ func (e *Engine) ReloadConfig(cfg Config) {
...
 }
 
+// List returns all snapshots available.
+func (e *Engine) ListMeta() ([]*tmtypes.Snapshot, error) {
+	e.avlLock.Lock()
+	defer e.avlLock.Unlock()
+
+	snapshots := make([]*tmtypes.Snapshot, 0, len(e.versions))
+	// TM list of snapshots is limited to the 10 most recent ones.
+	i := len(e.versions) - 11
```
Comment (ze97286) [inline core/snapshot/engine.go:229]: I know it's copied from the list() but why the -11?

Comment (jeremyletang) [reply core/snapshot/engine.go:229]: no idea, I would have done -10, but I guess there's some black magic reason for it to be -11?

Comment (ze97286) [reply core/snapshot/engine.go:229]: it must be because it's the smallest two-digit prime number (◔_◔)

Comment (jeremyletang) [reply core/snapshot/engine.go:229]: it must definitely be the reason.

**Pull request merged**